packages/opentelemetry-resource-detector-aws/src/detectors/AwsEc2Detector.ts

```diff
@@ -51,14 +53,18 @@ class AwsEc2Detector implements Detector {
         region,
         availabilityZone,
       } = await this._awsMetadataAccessor();
-      return new Resource({
-        [CLOUD_RESOURCE.PROVIDER]: 'aws',
-        [CLOUD_RESOURCE.ACCOUNT_ID]: accountId,
-        [CLOUD_RESOURCE.REGION]: region,
-        [CLOUD_RESOURCE.ZONE]: availabilityZone,
-        [HOST_RESOURCE.ID]: instanceId,
-        [HOST_RESOURCE.TYPE]: instanceType,
-      });
+      const resource = new Resource(
+        {
+          [CLOUD_RESOURCE.PROVIDER]: 'aws',
+          [CLOUD_RESOURCE.ACCOUNT_ID]: accountId,
+          [CLOUD_RESOURCE.REGION]: region,
+          [CLOUD_RESOURCE.ZONE]: availabilityZone,
+          [HOST_RESOURCE.ID]: instanceId,
+          [HOST_RESOURCE.TYPE]: instanceType,
+        },
+      );
+      config.logger.debug(`${this.constructor.name} found resource.`);
```

Comment:
Success and failure could be logged by the caller, which would make it consistent across detectors.

Reply:
@dyladan Moved in b295d8d but required to remove try/catch from aws detector. ptal